packages/flame/lib/src/components/mixins/has_hitboxes.dart

```diff
@@ -1,12 +1,38 @@
 import 'dart:collection';
-import 'dart:ui';
+import 'dart:math';
 
+import 'package:flutter/cupertino.dart';
+
+import '../../../components.dart';
 import '../../../extensions.dart';
-import '../../geometry/shape.dart';
-import '../position_component.dart';
+import '../../../geometry.dart';
 
 mixin HasHitboxes on PositionComponent {
   final List<HitboxShape> _hitboxes = <HitboxShape>[];
+  final Aabb2 _aabb = Aabb2();
+  bool _validAabb = false;
+  final Vector2 _maxHalfExtents = Vector2.zero();
+
+  @override
+  @mustCallSuper
+  void update(double dt) {
+    super.update(dt);
+    _validAabb = false;
+  }
+
+  Aabb2 get aabb {
+    if (_validAabb) {
+      return _aabb;
+    }
+    final size = scaledSize;
+    // This has +1 since a point on the edge of the bounding box is currently
+    // counted as inside.
+    final maxHalfExtent = (max(size.x, size.y) / 2) + 1;
+    _maxHalfExtents.setValues(maxHalfExtent, maxHalfExtent);
+    _aabb.setCenterAndHalfExtents(absoluteCenter, _maxHalfExtents);
```

Comment:
What if the component is rotated/scaled?

Reply:
That is what line 30 is for, it grows the box to the maximum that it should have to cover.
We could also get the absolute angle and "rotate" the AABB, I will see what is cheaper later.

Reply:
For example, if the component is a square (size.x == size.y), and rotated 45º, then its bounding rect should be sqrt(2) times larger than the size.

Reply:
Ah, true, good catch. I was meaning to take the length of the size/2.

Reply:
But then just rotating the AABB is probably cheaper anyways, so I'll do that.